experiments/src/browsergym/experiments/loop.py

```diff
@@ -180,17 +193,37 @@ def run(self):
             self.err_msg = err_msg
             self.stack_trace = stack_trace
 
-            logging.warning(err_msg + "\n" + stack_trace)
+            logger.warning(err_msg + "\n" + stack_trace)
             if _is_debugging() and self.enable_debug:
                 raise
 
         finally:
             try:
                 step_info.save_step_info(self.exp_dir)
+            except Exception as e:
+                logger.error(f"Error while saving step info in the finally block: {e}")
+            try:
                 _save_summary_info(episode_info, self.exp_dir, err_msg, stack_trace)
+            except Exception as e:
+                logger.error(f"Error while saving summary info in the finally block: {e}")
+            try:
                 env.close()
             except Exception as e:
-                logging.error(f"Error while finalizing the experiment loop: {e}")
+                logger.error(f"Error while closing the environment in the finally block: {e}")
+
+    def get_logger(self):
```

Comment:
This method doesn't seem necessary, it's only called in one place and is not meant to be called externally. A quick fix is to underscore it to make this explicit.
```suggestion
    def _get_logger(self):
```